### DISCUSSION TITLE
Limit access by email domain?

### DISCUSSION THREAD

**Discussion by parasiteoflife** [casdoor/casdoor#3460] [Q&A] [answered]
Can we limit access by email domain? Like I enable several providers but after the user login in using the provider I want casdoor to check the email domain and deny access if it's not in the allowed list, we can do this with Authentik with the flow policies, can we do this in Casdoor? How?

**Comment by hsluoyz** [comment]
@parasiteoflife can you provide a demo video about the flow "we can do this with Authentik with the flow policies" ?

**Reply by parasiteoflife** [reply]
I don't think a video will help explain this.

Here is the doc about it: https://version-2024-10.goauthentik.io/docs/customize/policies/working_with_policies/whitelist_email  

If we apply that policy, when the user tries to sign in or register, Authentik checks for the policies, if the user's email domain is not in the list they get an access denied error and are not signed in nor redirected to the service. I would like to have the same with casdoor, I don't want everybody to be able to register/sign in to the services or even to casdoor itself, only allowed domains or even allowed email addresses, can this be done?

**Reply by hsluoyz** [reply]
@parasiteoflife issue created here: https://github.com/casdoor/casdoor/issues/3461

**Reply by hsluoyz** [reply, marked as answer]
Fixed by: https://github.com/casdoor/casdoor/pull/3465
